packages/text-input/src/TextInput/TextInput.tsx

```diff
@@ -152,109 +123,37 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
+        errorMessage={errorMessage}
+        state={state}
+        size={sizeVariant}
+        disabled={disabled}
+        baseFontSize={baseFontSize}
+        darkMode={darkMode}
+        className={className}
+        id={id}
+        optional={optional}
       >
-        {(label || description) && (
-          <div className={textContainerStyle}>
-            {label && (
-              <Label
-                darkMode={darkMode}
-                htmlFor={id}
-                disabled={disabled}
-                className={inheritTypeScale}
-              >
-                {label}
-              </Label>
-            )}
-            {description && (
-              <Description
-                darkMode={darkMode}
-                disabled={disabled}
-                className={inheritTypeScale}
-              >
-                {description}
-              </Description>
-            )}
-          </div>
-        )}
-        <div className={inputContainerStyle}>
+        <FormFieldInputContainer>
           <input
             {...rest}
             aria-labelledby={ariaLabelledby}
             type={type}
-            className={cx(
-              baseInputStyle,
-              inputModeStyles[theme],
-              inputSizeStyles[sizeVariant],
-              inputStateStyles[state][theme],
-              inputFocusStyles[theme], // Always show focus styles
-              {
-                [inputPaddingForIndicator[sizeVariant]]: state !== State.None,
-                [inputPaddingForOptionalText[sizeVariant]]:
-                  shouldRenderOptionalText,
-              },
-            )}
             value={value}
             required={!optional}
             disabled={disabled}
             placeholder={placeholder}
             onChange={onValueChange}
             onBlur={onBlurHandler}
             ref={forwardRef}
-            id={id}
             autoComplete={disabled ? 'off' : rest?.autoComplete || 'on'}
             aria-invalid={state === 'error'}
           />
-
-          <div
-            className={cx(
-              iconClassName,
-              inputIndicatorStyle,
-              inputIndicatorSizeStyle[sizeVariant],
-            )}
-          >
-            {/* Render State Icon or Optional text*/}
-            {state === State.Valid && (
-              <CheckmarkIcon
-                role="presentation"
-                className={stateIndicatorStyles.valid[theme]}
-              />
-            )}
-
-            {state === State.Error && (
-              <WarningIcon
-                role="presentation"
-                className={stateIndicatorStyles.error[theme]}
-              />
-            )}
-
-            {shouldRenderOptionalText && (
-              <div
-                className={cx(
-                  optionalTextBaseStyle,
-                  optionalTextThemeStyle[theme],
-                )}
-              >
-                <p>Optional</p>
-              </div>
-            )}
```

Comment:
It looks like the `optional` text styles flash for a little bit on initial render

Reply:
Nvm - this happens on our current Storybook deployment too